doc/plugins.md

```diff
@@ -88,6 +88,8 @@ The list of plugins:
 *   ⚠️ [`remark-custom-blocks`](https://github.com/zestedesavoir/zmarkdown/tree/HEAD/packages/remark-custom-blocks#readme)
     — new syntax for custom blocks (new node types, rehype compatible)
     (👉 **note**: [`remark-directive`][d] is similar and up to date)
+*   🟢 [`remark-d2`](https://github.com/mech-a/remark-d2)
+    — turn [d2](https://d2lang.com/) diagram code blocks into images
```

Comment:
Hi! Some tips on how to improve your plugin:

- You don’t need `'use strict'` in ESM, they are always strict
- Instead of using `node:path`, I recommend using URLs, they work across platforms (unix vs windows)
- Throw errors instead of printing to `console.error` when things are misconfigured
- You don’t need to `path.parse(file.path)`, `file` already has `direname` and other fields
- replace `process.cwd()` with `file.cwd`, files can theoretically exist with multiple, different, “`cwd`s”
- Use async versions of `fs` calls, the transformer can be async
- an `exists` call before a `mkdir` or `write` call is an anti-pattern, instead, try to `mkdir` or `write`, and swallow the error if it already exists
- do not change `node.type`, instead, replace one node with another, some more info here: https://unifiedjs.com/learn/recipe/remove-node/#removing-a-node
- It is probably better to do async work instead of `spawn`, so that when many files are processed, or many code blocks exist, work can be done at the same time
- What is the benefit of the HTML version? What can it do more than the markdown version? If there are no benefits, I recommend removing it, because raw strings of HTML don’t always work well with markdown, particularly when dealing with VDOM frameworks like React, such as MDX. There are also perhaps openings for XSS attacks, or at least bugs, when building a string of HTML like you do now, instead of using something like [hastscript](https://github.com/syntax-tree/hastscript) and [hast-util-to-html](https://github.com/syntax-tree/hast-util-to-html)!

Reply:
Thank you for this feedback. I'll integrate your feedback where applicable and reply back.

Reply:
how’s it going with the feedback?

Reply:
Ping! :)